client/my-sites/marketing/connections/bluesky.tsx

```diff
@@ -0,0 +1,159 @@
+import { FormEvent, useEffect, useId, useRef, useState } from 'react';
+import FormsButton from 'calypso/components/forms/form-button';
+import FormSettingExplanation from 'calypso/components/forms/form-setting-explanation';
+import { Connection, Service } from './types';
+
+interface Props {
+	service: Service;
+	action: () => void;
+	connectAnother: () => void;
+	connections: Connection[];
+	isConnecting: boolean;
+}
+
+/**
+ * Example valid handles:
+ * - username.bsky.social
+ * - user-name.bsky.social
+ * - my_domain.com.bsky.social
+ * - my-domain.com.my-own-server.com
+ * @param {string} handle - Handle to validate
+ * @returns {boolean} - Whether the handle is valid
+ */
+function isValidBlueskyHandle( handle: string ) {
+	const parts = handle.split( '.' ).filter( Boolean );
+
+	// A valid handle should have at least 3 parts - username, domain, and tld
+	if ( parts.length < 3 ) {
+		return false;
+	}
+
+	return parts.every( ( part ) => /^[a-z0-9_-]+$/i.test( part ) );
+}
+
+const isAlreadyConnected = ( connections: Array< Connection >, handle: string ) => {
+	return connections.some( ( connection ) => {
+		const { external_display } = connection;
+		return external_display === handle;
+	} );
+};
+
+export const Bluesky: React.FC< Props > = ( {
+	service,
+	action,
+	connectAnother,
+	connections,
+	isConnecting,
+} ) => {
+	const translate = useTranslate();
+	const [ error, setError ] = useState( '' );
+	const formRef = useRef< HTMLFormElement >( null );
+
+	// After sucessfully connecting an account, reset the handle.
+	// Disabled react-hooks/exhaustive-deps because we don't want to run this on handle change
+	useEffect( () => {
+		const handle = formRef.current?.elements.namedItem( 'handle' ) as HTMLInputElement;
+
+		if ( ! isConnecting && isAlreadyConnected( connections, handle.value ) ) {
+			formRef.current?.reset();
+		}
+	}, [ isConnecting, connections ] ); // eslint-disable-line react-hooks/exhaustive-deps
+
+	/**
+	 * Handle the Connect account submission.
+	 */
+	const handleSubmit = ( e: FormEvent< HTMLFormElement > ) => {
+		e.preventDefault();
+		e.stopPropagation();
+
+		const formData = new FormData( e.target as HTMLFormElement );
+
+		// Let us make the user's life easier by removing the leading "@" if they added it
+		const handle = ( formData.get( 'handle' )?.toString().trim() || '' ).replace( /^@/, '' );
+		const app_password = formData.get( 'app_password' )?.toString().trim() || '';
+
+		if ( isAlreadyConnected( connections, handle ) ) {
+			return setError( translate( 'This account is already connected.' ) );
+		}
+
+		if ( ! handle || ! isValidBlueskyHandle( handle ) ) {
+			return setError( translate( 'Please enter a valid handle.' ) );
+		}
+
+		const url = new URL( service.connect_URL );
+		url.searchParams.set( 'handle', handle );
+		url.searchParams.set( 'app_password', app_password );
+
+		// TODO: Fix this to avoid mutating props
+		service.connect_URL = url.toString();
```

Comment:
I hate myself for doing this. We should fix this in a follow. It was done previously. Reported here https://github.com/Automattic/wp-calypso/pull/73845#discussion_r1778471553